services/frontend-service/src/ui/components/SideBar/SideBar.tsx

```diff
@@ -15,11 +15,167 @@ along with kuberpult.  If not, see <http://www.gnu.org/licenses/>.
 
 Copyright 2021 freiheit.com*/
 import { Button } from '../button';
-import { HideBarWhite } from '../../../images';
+import { DeleteGray, HideBarWhite } from '../../../images';
+import { BatchAction } from '../../../api/api';
+import { deleteAction, useActions } from '../../utils/store';
+import { useCallback } from 'react';
+
+export enum ActionTypes {
+    Deploy,
+    PrepareUndeploy,
+    Undeploy,
+    DeleteQueue,
+    CreateEnvironmentLock,
+    DeleteEnvironmentLock,
+    CreateApplicationLock,
+    DeleteApplicationLock,
+    UNKNOWN,
+}
+
+type ActionDetails = {
+    type: ActionTypes;
+    name: string;
+    summary: string;
+    dialogTitle: string;
+    description?: string;
+
+    // action details optional
+    environment?: string;
+    application?: string;
+    lockId?: string;
+    lockMessage?: string;
+    version?: number;
+};
+
+const getActionDetails = ({ action }: BatchAction): ActionDetails => {
+    switch (action?.$case) {
+        case 'createEnvironmentLock':
+            return {
+                type: ActionTypes.CreateEnvironmentLock,
+                name: 'Create Env Lock',
+                dialogTitle: 'Are you sure you want to add this environment lock?',
+                summary: 'Create new environment lock on ' + action.createEnvironmentLock.environment,
+                environment: action.createEnvironmentLock.environment,
+            };
+        case 'deleteEnvironmentLock':
+            return {
+                type: ActionTypes.DeleteEnvironmentLock,
+                name: 'Delete Env Lock',
+                dialogTitle: 'Are you sure you want to delete this environment lock?',
+                summary: 'Delete environment lock on ' + action.deleteEnvironmentLock.environment,
```

Comment:
please add the message of the lock here to the summary
'Delete environement lock (' + action.message + .... on ...